### ISSUE TITLE
JavadocMethod: generic type is not recognized

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#2409]
```
/var/tmp $ java -jar checkstyle-6.11.2-all.jar -c 3.xml 3.java
Starting audit...
/var/tmp/3.java:8:14: error: Expected @param tag for '<R>'.
/var/tmp/3.java:8:17: error: Expected @param tag for '<T>'.
Audit done.
Checkstyle ends with 2 errors.

/var/tmp $ cat 3.java 
public abstract class Shell<S> {
  /**
   * Does something.
   * @param callable This is a param.
   * @return This is a result.
   * @throws IOException This is an exception.
   */
  protected <R, T extends Foo<S> & Callable<? extends R>> R bar(final T callable)
      throws IOException {
    return null;
  }
}

/var/tmp $ cat 3.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://jee-qa/jee-qa/trunk/checkstyle/configuration_1_2.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocMethod" />
    </module>
</module>
```

Expected: no warnings.

**Comment by mkordas** [comment]
@romani, why no warnings are expected here?

**Comment by romani** [comment]
I do not see what violation should be there, please point to it

**Comment by mkordas** [comment]
I'd expect just:
```
/var/tmp/3.java:8:14: error: Expected @param tag for '<R>'.
/var/tmp/3.java:8:17: error: Expected @param tag for '<T>'.
```
Source: http://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@param

**Comment by romani** [comment]
yes , you are right, Check works correctly.

```
/var/tmp/1 $ cat Shell.java
public abstract class Shell<S> {
  /**
   * Does something.
   * @param callable This is a param.
   * @return This is a result.
   */
  protected <R, T extends Comparable<S> & Comparable<? extends R>> R bar(final T callable) {
    return null;
  }
}

/var/tmp/1 $ javadoc Shell.java
Loading source file Shell.java...
Constructing Javadoc information...
Standard Doclet version 1.8.0_45
Building tree for all the packages and classes...
Generating ./Shell.html...
Shell.java:7: warning: no @param for <R>
  protected <R, T extends Comparable<S> & Comparable<? extends R>> R bar(final T callable) {
                                                                     ^
Shell.java:7: warning: no @param for <T>
  protected <R, T extends Comparable<S> & Comparable<? extends R>> R bar(final T callable) {
          
```
